python/cudf_polars/cudf_polars/containers/dataframe.py

```diff
@@ -40,6 +40,20 @@ def _create_polars_column_metadata(
     return plc.interop.ColumnMetadata(name=name, children_meta=children_meta)
 
 
+# This is also defined in pylibcudf.interop
+class _ObjectWithArrowMetadata:
```

Comment:
I think https://github.com/rapidsai/cudf/issues/18722 is the primary limitation that necessitates this workaround cc @vyasr

Reply:
Yes, exactly. Without this you lose the underlying struct fields. This solution is fine for now while we sort out #18722.